### PR TITLE
5227 handle token fetch error

### PR DESCRIPTION
**Pull request by tcobbs-bentley**
Fix for issue: https://github.com/iTwin/itwinjs-core/issues/5227

### CODE REVIEW

**Comment by tcobbs-bentley** [comment]
@Mergifyio backport release/3.7.x

**Comment by tcobbs-bentley** [comment]
@calebmshafer This fixes a pretty severe bug. (If a mobile user canceled signin for any reason, they had to kill and restart the app before they could sign in again.) I put in a mergify bot comment for 3.7, but I don't know if it is configured to listen to me. Also, if 3.7 isn't released soon, then SYNCHRO Field will need this to be backported to 3.6.

**Comment by calebmshafer** [comment]
> @calebmshafer This fixes a pretty severe bug. (If a mobile user canceled signin for any reason, they had to kill and restart the app before they could sign in again.) I put in a mergify bot comment for 3.7, but I don't know if it is configured to listen to me. Also, if 3.7 isn't released soon, then SYNCHRO Field will need this to be backported to 3.6.

@tcobbs-bentley sounds good. 3.7 is planned for early next week so we should be fine to just put it there.

File: core/mobile/src/frontend/MobileAuthorizationFrontend.ts [hunk 1]
```diff
@@ -25,19 +25,28 @@ export class MobileAuthorizationFrontend implements AuthorizationClient {
...
-      this._accessToken = result[0];
-      if (result[1])
-        this._expirationDate = new Date(result[1]);
-      this._fetchingToken = false;
-      return this._accessToken;
+      try {
+        this._fetchingToken = true;
+        const result = await MobileApp.callBackend("getAccessToken");
+        this._accessToken = result[0];
+        if (result[1])
```
Comment (toddsouthenbentley) [inline core/mobile/src/frontend/MobileAuthorizationFrontend.ts:40]: I would prefer this more compact code:
`this._expirationDate = result[1] ? new Date(result[1]) : undefined;`

Comment (tcobbs-bentley) [reply core/mobile/src/frontend/MobileAuthorizationFrontend.ts]: Fixed.

**Review by toddsouthenbentley — APPROVED** [review]

**Review by pmconne — APPROVED** [review]

**Pull request merged**